airbyte-webapp/src/components/connectorBuilder/Builder/BuilderSidebar.tsx

```diff
@@ -0,0 +1,157 @@
+    </button>
+  );
+};
+
+interface StreamSelectButtonProps {
+  streamNum: number;
+  onSelectStream: (streamNum: number, streamName: string) => void;
+  selected: boolean;
+}
+
+const StreamSelectButton: React.FC<StreamSelectButtonProps> = ({ streamNum, onSelectStream, selected }) => {
+  const streamPath = `streams[${streamNum}]`;
+  const [field] = useField(`${streamPath}.name`);
+
+  return (
+    <ViewSelectButton selected={selected} onClick={() => onSelectStream(streamNum, field.value)}>
+      {field.value}
+    </ViewSelectButton>
+  );
+};
+
+interface BuilderSidebarProps {
+  className?: string;
+  toggleYamlEditor: () => void;
+  numStreams: number;
+  onViewSelect: (selected: BuilderView, streamName?: string) => void;
+  selectedView: BuilderView;
+}
+
+export const BuilderSidebar: React.FC<BuilderSidebarProps> = ({
+  className,
+  toggleYamlEditor,
+  numStreams,
+  onViewSelect,
+  selectedView,
+}) => {
+  const { openConfirmationModal, closeConfirmationModal } = useConfirmationModalService();
+  const { yamlManifest } = useConnectorBuilderState();
+  const { setValues } = useFormikContext();
+  const handleResetForm = () => {
+    openConfirmationModal({
+      text: "connectorBuilder.resetModal.text",
+      title: "connectorBuilder.resetModal.title",
+      submitButtonText: "connectorBuilder.resetModal.submitButton",
+      onSubmit: () => {
+        setValues(DEFAULT_BUILDER_FORM_VALUES);
+        onViewSelect("global");
+        closeConfirmationModal();
+      },
+    });
+  };
+
+  const [field] = useField("global.connectorName");
+
+  return (
+    <div className={classnames(className, styles.container)}>
+      <UiYamlToggleButton yamlSelected={false} onClick={toggleYamlEditor} />
+
+      {/* TODO: replace with uploaded img when that functionality is added */}
+      <img className={styles.connectorImg} src="/logo.png" alt="Connector Logo" />
+
+      <div className={styles.connectorName}>
+        <Heading as="h2" size="sm" className={styles.connectorNameText}>
+          {field.value}
+        </Heading>
+      </div>
+
+      <ViewSelectButton
+        className={styles.globalConfigButton}
+        selected={selectedView === "global"}
+        onClick={() => onViewSelect("global")}
+      >
+        <FontAwesomeIcon icon={faSliders} />
+        <FormattedMessage id="connectorBuilder.globalConfiguration" />
+      </ViewSelectButton>
+
+      <div className={styles.streamsHeader}>
+        <Text className={styles.streamsHeading} size="xs" bold>
+          <FormattedMessage id="connectorBuilder.streamsHeading" values={{ number: numStreams }} />
+        </Text>
+
+        <AddStreamButton
+          className={styles.addStreamButton}
+          numStreams={numStreams}
+          onAddStream={(addedStreamNum, addedStreamName) => onViewSelect(addedStreamNum, addedStreamName)}
+        />
+      </div>
+
+      <div className={styles.streamList}>
+        {Array.from(Array(numStreams).keys()).map((streamNum) => {
```

Comment:
This structure with numStreams seems more complex than necessary - as you don't access anything out of the stream except for the value of name anyway, it seems easier to do a `useField("streams")` here in the sidebar and then simply map it here like this:
```js
streams.map(stream => (<StreamSelectButton stream={stream} {/* ... */} />))
```

Reply:
Good suggestion - I actually was able to simplify this even further and completely eliminate the `StreamSelectButton` component based on your suggestion